Soundex.Tests.cs

```diff
@@ -2,17 +2,80 @@
 
 public class SoundexTests
 {
-    [Fact]
-    public void HandlesEmptyString()
-    {
-        Assert.Equal(string.Empty, Soundex.GenerateSoundex(""));
-    }
-
-    [Fact]
-    public void HandlesSingleCharacter()
-    {
-        Assert.Equal("A000", Soundex.GenerateSoundex("A"));
-    }
-
-   
+  [Fact]
+  public void GenerateSoundex_EmptyString_ReturnsEmptyString()
+  {
+    string name = "";
+    string expectedSoundex = "";
+
+    string actualSoundex = Soundex.GenerateSoundex(name);
+
+    Assert.Equal(expectedSoundex, actualSoundex);
+  }
+
+  [Fact]
+  public void GenerateSoundex_NullInput_ReturnsEmptyString()
+  {
```

Comment:
it appears that code throws runtime error ! check and fix it

Reply:
ok venu still working on it